eventbroadcaster/notifiersink.go

```diff
@@ -29,36 +28,100 @@ type NotifierSink struct {
 	logger   *zap.SugaredLogger
 }
 
-// RegisterNotifierSinks builds notifiers sinks and registers them on the given broadcaster
-func RegisterNotifierSinks(mgr ctrl.Manager, broadcaster record.EventBroadcaster, notifiersConfig eventnotifier.NotifiersConfig, logger *zap.SugaredLogger) (err error) {
-	client := mgr.GetClient()
-
-	notifiers, err := eventnotifier.GetNotifiers(notifiersConfig, logger)
-
+// RegisterNotifierSinks registers notifiers sinks on the given broadcaster
+func RegisterNotifierSinks(mgr ctrl.Manager, broadcaster record.EventBroadcaster, notifiers []eventnotifier.Notifier, logger *zap.SugaredLogger) {
 	for _, notifier := range notifiers {
 		logger.Infof("notifier %s enabled", notifier.GetNotifierName())
 
-		broadcaster.StartRecordingToSink(&NotifierSink{client: client, notifier: notifier, logger: logger})
+		broadcaster.StartRecordingToSink(&NotifierSink{client: mgr.GetClient(), notifier: notifier, logger: logger})
 	}
 
 	corev1Client, _ := corev1client.NewForConfig(mgr.GetConfig())
 
 	broadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: corev1Client.Events("")})
-
-	return
 }
 
 func (s *NotifierSink) Create(event *corev1.Event) (*corev1.Event, error) {
-	dis, err := s.getDisruption(event)
-	if err != nil {
+	s.logger.Debugw("CREATE event received:", "event", event)
+
+	var obj client.Object
+
+	switch event.InvolvedObject.Kind {
+	case v1beta1.DisruptionKind:
+		disruption, err := s.getDisruption(event)
+		if err != nil {
+			s.logger.Warn(err)
+
+			disruption = v1beta1.Disruption{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       event.InvolvedObject.Kind,
+					APIVersion: event.InvolvedObject.APIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:                       event.InvolvedObject.Name,
+					GenerateName:               event.GetGenerateName(),
+					Namespace:                  event.InvolvedObject.Namespace,
+					UID:                        event.InvolvedObject.UID,
+					ResourceVersion:            event.InvolvedObject.ResourceVersion,
+					Generation:                 event.GetGeneration(),
+					CreationTimestamp:          event.GetCreationTimestamp(),
```

Comment:
I suspect most of the `event.GetNoun()` functions are going to give you metadata about the _Event_, not about the involvedObject, which is what we're treating them as.

We should stick to getting whatever we can out of the InvolvedObject, and leave the rest unset. The receivers of these http events can then handle that however they feel is best. But we should never accidentally send wrong data, better to just send incomplete data

Reply:
Good point, those data relate to the event itself and not the involved object. Thanks a lot of spot this bug 🙏